zipline/pipeline/loaders/blaze.py

```diff
@@ -648,11 +648,31 @@ def overwrite_from_dates(asof, dense_dates, sparse_dates, asset_idx, value):
     -------
     overwrite : Float64Overwrite
         The overwrite that will apply the new value to the data.
+
+    Notes
+    -----
+    The overwrite is forward filled in the sense that it applies from the index
+    of the ``asof`` in the ``dense_dates`` to the index of the next date in the
+    ``sparse_dates`` in the ``dense_dates``.
+
+    For example:
+    let ``asof = pd.Timestamp('2014-01-02')``,
+        ``dense_dates = pd.date_range('2014-01-01', '2014-01-05')``
+        ``sparse_dates = pd.to_datetime(['2014-01', '2014-02', '2014-04'])``
+
+    Then the overwrite will apply to indexes: 1, 2, 3, 4
     """
     first_row = dense_dates.searchsorted(asof)
-    last_row = dense_dates.searchsorted(
-        sparse_dates[sparse_dates.searchsorted(asof, 'right')],
-    ) - 1
+    next_idx = sparse_dates.searchsorted(asof, 'right')
+    if next_idx == len(sparse_dates):
```

Comment:
This is forward-filling "all dense dates that are between the as-of dates and the next sparse date after the asof_date".